packages/millicast-sdk/src/workers/TransformWorker.worker.js

```diff
@@ -1,44 +1,78 @@
 import { addH26xSEI, extractH26xMetadata } from '../utils/Codecs'
 
-let uuid = ''
-let payload = ''
+const DROPPED_SOURCE_TIMEOUT = 2000
+const metadata = []
 let codec = ''
 let codecMap = {}
+const synchronizationSources = {}
+let synchronizationSourcesWithMetadata = []
 
-function createReceiverTransform () {
+function createReceiverTransform (mid) {
   return new TransformStream({
     start () {},
     flush () {},
     async transform (encodedFrame, controller) {
-      const frameCodec = codecMap[encodedFrame.getMetadata().payloadType] || codec.toUpperCase()
-      if (frameCodec === 'H264') {
-        const metadata = extractH26xMetadata(encodedFrame, frameCodec)
-        if (metadata.timecode || metadata.unregistered || metadata.seiPicTimingTimeCodeArray?.length > 0) {
-          self.postMessage({ metadata })
+      // eslint-disable-next-line no-undef
+      if (encodedFrame instanceof RTCEncodedVideoFrame) {
+        const frameCodec = codecMap[encodedFrame.getMetadata().payloadType] || codec.toUpperCase()
+        if (frameCodec === 'H264') {
+          const metadata = extractH26xMetadata(encodedFrame, frameCodec)
+          if (metadata.timecode || metadata.unregistered || metadata.seiPicTimingTimeCodeArray?.length > 0) {
+            self.postMessage({ mid, metadata })
+          }
         }
       }
       controller.enqueue(encodedFrame)
     }
   })
 }
 
+function clearMetadata () {
+  if (Object.keys(synchronizationSources).sort().join() === synchronizationSourcesWithMetadata.sort().join()) {
+    metadata.shift()
+    synchronizationSourcesWithMetadata = []
+  }
+}
+
+function refreshSynchronizationSources (newSyncSource) {
+  const now = new Date().getTime()
+  synchronizationSources[newSyncSource] = now
+
+  const sourcesToDelete = Object.keys(synchronizationSources)
+    .filter(source => (now - synchronizationSources[source] > DROPPED_SOURCE_TIMEOUT))
```

Comment:
How do we get the time out value as 2 seconds? Any industry standard requirement?

Reply:
No standard found, it was a number we thought we could make sure the resolution dropped. Any suggestions are welcome

Reply:
So the reason this function exists is to ensure that the `synchronisationSources` Map doesn't keep growing if sources keep getting added/removed from the list, isn't it? It's to ensure faster lookups

Reply:
I'm wondering if we need to store the timestamp at all, and if it can just be an array of `ssrc`. What's the implication of that. 

@vincentsong what do you think?